contracts/strategies/JoeVoter.sol

```diff
@@ -0,0 +1,141 @@
+
+    modifier onlyJoeVoterProxy() {
+        require(msg.sender == voterProxy, "JoeVoter::onlyJoeVoterProxy");
+        _;
+    }
+
+    constructor(address _owner, address _stakingContract) ERC20("Yield Yak JOE", "yyJOE") {
+        stakingContract = IVeJoeStaking(_stakingContract);
+        transferOwnership(_owner);
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice veJOE balance
+     * @return uint256 balance
+     */
+    function veJOEBalance() external view override returns (uint256) {
+        return veJOE.balanceOf(address(this));
+    }
+
+    /**
+     * @notice Enable/disable deposits
+     * @dev Restricted to owner
+     * @param newValue bool
+     */
+    function updateDepositsEnabled(bool newValue) external onlyOwner {
+        require(depositsEnabled != newValue);
+        depositsEnabled = newValue;
+    }
+
+    /**
+     * @notice External deposit function for JOE
+     * @param _amount to deposit
+     */
+    function deposit(uint256 _amount) external {
+        require(depositsEnabled == true, "JoeVoter:deposits disabled");
+        require(IERC20(JOE).transferFrom(msg.sender, address(this), _amount), "JoeVoter::transfer failed");
+        _deposit(_amount);
+    }
+
+    /**
+     * @notice Update VeJoeStaking address
+     * @param _stakingContract new address
+     */
+    function setStakingContract(address _stakingContract) external override onlyOwner {
+        stakingContract = IVeJoeStaking(_stakingContract);
+    }
+
+    /**
+     * @notice Update proxy address
+     * @dev Very sensitive, restricted to owner
+     * @param _voterProxy new address
+     */
+    function setVoterProxy(address _voterProxy) external override onlyOwner {
+        voterProxy = _voterProxy;
+    }
+
+    /**
+     * @notice Update veJOE balance
+     * @dev Any one may call this
+     */
+    function claimVeJOE() external override {
+        stakingContract.claim();
+    }
+
+    /**
+     * @notice Deposit function for JOE
+     * @dev Restricted to proxy
+     * @param _amount to deposit
+     */
+    function depositFromBalance(uint256 _amount) external override onlyJoeVoterProxy {
+        require(depositsEnabled == true, "JoeVoter:deposits disabled");
+        _deposit(_amount);
+    }
+
+    /**
+     * @notice Deposits JOE and mints yyJOE at 1:1 ratio
+     * @param _amount to deposit
+     */
+    function _deposit(uint256 _amount) internal {
+        IERC20(JOE).approve(address(stakingContract), _amount);
+        _mint(msg.sender, _amount);
+        stakingContract.deposit(_amount);
+        IERC20(JOE).approve(address(stakingContract), 0);
+    }
+
+    /**
+     * @notice Helper function to wrap AVAX
+     * @return amount wrapped to WAVAX
+     */
+    function wrapAvaxBalance() external override onlyJoeVoterProxy returns (uint256) {
+        uint256 balance = address(this).balance;
+        if (balance > 0) {
+            WAVAX.deposit{value: balance}();
+        }
+        return balance;
```

Comment:
would here be a better idea to use WAVAX.balanceOf? this way if we need to fix something we can airdrop WAVAX

Reply:
This method is only there if there are ever native Avax rewards again. I already put it in because I couldn't find a way to do it with execute.